File: c++/Memory.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <string.h>
#include "Product.cpp"

using namespace std;

class Memory : public Product
    {
        private:
            //atribut
            string freq;
            string memSize;
            string spCuda;
        
        public:
            //konstruktor
            Memory(){
            }

            //get dan set product

            void setFreq(string freq){
                this->freq = freq;
            }

            string getFreq(){
                return this->freq;
            }

            void setMemSize(string memSize){
                this->memSize = memSize;
            }

            string getMemSize(){
                return this->memSize;
            }

            void setSpCuda(string spCuda){
                this->spCuda = spCuda;
            }

            string getSpCuda(){
                return this->spCuda;
            }

            //destructor
            ~Memory(){
            }
            
    };

File: java/Memory.java
public class Memory extends Product{

    private String freq;
    private String memSize;
    private String spCuda;

    Memory(){
    }

    //get dan set Memory

    public void setFreq(String freq){
        this.freq = freq;
    }

    public String getFreq(){
        return this.freq;
    }

    public void setMem_size(String memSize){
        this.memSize = memSize;
    }

    public String getMem_size(){
        return this.memSize;
    }

    public void setSp_cuda(String spCuda){
        this.spCuda = spCuda;
    }

    public String getSp_cuda(){
        return this.spCuda;
    }

}
